rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/ProductVersionEndpoint.java

```diff
@@ -114,6 +114,23 @@ void update(
             @Parameter(description = PV_ID) @PathParam("id") int id,
             @NotNull ProductVersion productVersion);
 
+    @Operation(summary = "Patch an existing product version.",
+            responses = {
+                    @ApiResponse(responseCode = SUCCESS_CODE, description = SUCCESS_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ProductVersion.class))),
+                    @ApiResponse(responseCode = INVALID_CODE, description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(responseCode = NOT_FOUND_CODE, description = NOT_FOUND_DESCRIPTION),
```

Comment:
We use INVALID_CODE instead of NOT_FOUND for methods that take body as a parameter (like update).
Also you should add description to the `/Newcastle+Rest+API+Guidelines#NewcastleRestAPIGuidelines-CommonCRUDentityoperations` confluence doc.
And there should also probably be the CONFLICT code response.

Reply:
404 NOT_FOUND is returned by the GET method that is invoked before the PATCH one. To change the behaviour it requires to create a custom PatchMethodFilter which unfortunately cannot be overridden in current RestEasy version but requires to patch EAP distribution. I would leave it as is.